### PR TITLE
feat: rework gateway page and add recursive gateway

### PR DESCRIPTION
**Pull request by 2color**
# Describe your changes

- Clean up deprecated and unhelpful content
- Add recursive/non-recursive to gateway types
- Add trusted and trustless gateways to gateway types

## Checklist before merging
- [ ] Passing all required checks (The beta **Check Markdown links for modified files** check is not required)

### CODE REVIEW

File: docs/concepts/ipfs-gateway.md [hunk 1]
```diff
@@ -3,103 +3,120 @@ title: IPFS Gateway
...
-:::
+Public ([recursive](#recursive-vs-non-recursive-gateways)) gateways are provided by the IPFS various other organizations, including the [IPFS Foundation](./public-utilities.md#public-ipfs-gateways) as a public utility.
 
-## Gateway providers
- 
-Regardless of who deploys a gateway and where, any IPFS gateway resolves access to any requested IPFS [content identifier](content-addressing.md). Therefore, for best performance, when you need the service of a gateway, you should use the one closest to you.
+For a list of public gateways, see the [IPFS Gateways Checker](https://ipfs.fyi/gateways).
 
-### Your local gateway
+## Gateway types
```
Comment (2color) [inline docs/concepts/ipfs-gateway.md:46]: I wonder if it makes sense to add a "Gateway providers" type for non-recursive gateways

File: docs/concepts/ipfs-gateway.md [hunk 2]
```diff
@@ -3,103 +3,120 @@ title: IPFS Gateway
...
 _Read-only gateways_ are the simplest kind of gateway. This gateway type provides a way to fetch IPFS content using the HTTP GET method.
 
-### Authenticated gateways
+## Authenticated gateways
 
 If a gateway provider wants to limit access to requests with authentication, they may need to configure a reverse proxy, develop an IPFS plugin, or set a cache-layer above IPFS.
 
 Configuring a reverse proxy is the most popular way for providers handling authentication. Reverse proxy can also keep the original IPFS API calls which makes gateway adaptable to all IPFS SDK and toolkits.
 
 ![Auth with Reverse proxy](./images/ipfs-gateways/public-authed-gateway.png)
```
Comment (lidel) [inline docs/concepts/ipfs-gateway.md:79]: nit: this image is wrong and confusing (gateway has no `/api/v0`), just remove it.

```suggestion
```

File: docs/concepts/ipfs-gateway.md [hunk 3]
```diff
@@ -3,103 +3,120 @@ title: IPFS Gateway
...
 
-Providers can design their own centralized authentication service like [Infura IPFS Auth](https://docs.infura.io/networks/ipfs/how-to/authenticate-requests), or a decentralized authentication service like [IPFS W3Auth](https://wiki.crust.network/docs/en/buildIPFSWeb3AuthGW)).
+## Gateway request lifecycle
+
+:::callout
+This section uses the _default_ recursive gateway request lifecycle of [IPFS Kubo](https://github.com/ipfs/kubo) to introduce the basic concepts in the lifecycle. However, non-recursive gateways only serve content that they have and/or want to provide. For example, a Kubo gateway with `NoFetch` enabled will **not** attempt to retrieve content from the network.
```
Comment (lidel) [inline docs/concepts/ipfs-gateway.md:84]:
```suggestion
This section uses the _default_ recursive gateway request lifecycle of [IPFS Kubo](https://github.com/ipfs/kubo) to introduce the basic concepts in the lifecycle. However, non-recursive gateways only serve content that they have and/or want to provide. For example, a Kubo gateway with [`Gateway.NoFetch=true`](https://github.com/ipfs/kubo/blob/master/docs/config.md#gatewaynofetch) will **not** attempt to retrieve content from the network.
```

File: docs/concepts/ipfs-gateway.md [hunk 4]
```diff
@@ -3,103 +3,120 @@ title: IPFS Gateway
...
+:::
+
+When a client request for a CID reaches an IPFS gateway, the gateway first checks whether the CID is cached locally. At this point, one of the following occurs:
+
+- **If the CID is cached locally**, the gateway responds with the content referred to by the CID, and the lifecycle is complete.
+
+- **If the CID is not in the local cache**, the gateway will attempt to retrieve it from the network.
```
Comment (lidel) [inline docs/concepts/ipfs-gateway.md:91]: Good opportunity to show difference between both gw types?

```suggestion
- **If the CID is not in the local cache**, a non-recursive gateway would error, however our gateway is recursive and will attempt to retrieve it from the network.
```

File: docs/concepts/ipfs-gateway.md [hunk 5]
```diff
@@ -141,7 +158,7 @@ DNSLink resolution occurs when the gateway recognizes an IPNS identifier contain
...
+2. The gateway searches the DNS TXT records of the requested domain `{_dnslink.example.com}` for a string of the form `dnslink=/ipfs/{CID}`. If found, the gateway uses the specified CID to serve up `ipfs://{CID}/{optional path}`. As with path resolution, this form of DNSLink resolution violates the single-origin policy. The domain operator may ensure single-origin policy compliance — and the delivery of the current version of content — by adding an `Alias` record in the DNS that refers to a suitable IPFS gateway; e.g., `gateway.ipfs.io`.
 3. The `Alias` record redirects any access to that `example.com` to the specified gateway. Hence the browser's request to `https://{example.com}/{optional path to resource}` redirects to the gateway specified in the `Alias`.
 4. The gateway employs DNSLink resolution to return the current content version from IPFS.
 5. The browser does not perceive the gateway as the origin of the content and therefore enforces the single-origin policy to protect `example.com`.
```
Comment (lidel) [inline docs/concepts/ipfs-gateway.md:164]: Wow.. there is a lot of.. lets say untrue, controversial or vague statements in the old text here:
- We describe DNSLink name support on path gateway mixing it with logic and advanced setup based on `ALIAS` DNS record, and we are not mentioning `Host` header  case which is literally the core of https://specs.ipfs.tech/http-gateways/dnslink-gateway/
- Using alias is a gigantic footgun if you dont own both domains. It delegates authority to somneone, they can pretend to own your domain now. This is why Fleek uses CNAME for `_dnslink` subdomain, and nothing more.
- comment about origin isolation is incorrect (there is no isolation on path gateway, we protect noone)


I suspect this was glue together from various notes that made sense on their own, but in this form it is plain wrong.

Let's clean this up:

````suggestion
2. The gateway searches the DNS TXT records  on `_dnslink.` subdomain (`_dnslink.example.com`) for a string of the form `dnslink=/ipfs/{CID}`. If found, the gateway uses the specified content identifier to find and serve up `ipfs://{CID}/{optional path}`. 

It is possible to use HTTP gateway for serving content on the DNSLink domain itself:

1. Point `example.com` at IP of your HTTP gateway, make sure `A`/`AAAA`/`HTTPS` records are set, and TLS termination is configured.
2. Client sends request to:

   ```bash
   https://{example.com}/{optional path}
   ```

3. Gateway detects HTTP header `Host: example.com` in the incoming request and searches DNSLink the same way as in previous example.
````

**Review by lidel — CHANGES_REQUESTED** [review]
Thank you for cleaning this up @2color, I found some easy wins while reading, suggestions below.

Most important is DNSLink section cleanup. If suggestions look goot to you, feel free to merge.

File: docs/concepts/ipfs-gateway.md [hunk 6]
```diff
@@ -3,103 +3,119 @@ title: IPFS Gateway
...
 
-### Private gateways
+[Trustless, verifiable retrieval](../reference/http/gateway.md#trustless-verifiable-retrieval) from non-recursive gateways is becoming a popular way to provide IPFS content to the network ([HTTP](https://docs.ipfs.tech/reference/http/gateway/#trustless-verifiable-retrieval) as an alternative or in addition to [Bitswap](../concepts/glossary.md#bitswap)).
 
-_Private gateways_ are configured to limit access to requests from specific domains or parts of the public internet. 
+## Trusted vs. trustless gateways
 
-They are frequently, but not exclusively, used behind firewalls. Running [IPFS Desktop](https://github.com/ipfs-shipyard/ipfs-desktop#ipfs-desktop) or another form of IPFS node triggers connection attempts to other IPFS peers. Private network administrators may treat such connection attempts as potential security vulnerabilities. Private IPFS gateway servers located inside the private network and running a trusted code base provide an alternative architecture for read/write access to externally-hosted IPFS content.
+See [Trusted vs. Trustless Gateways](../reference/http/gateway/#trusted-vs-trustless) for more information.
```
Comment (lidel) [inline docs/concepts/ipfs-gateway.md:67]:
```suggestion
See [Trusted vs. Trustless Gateways](../reference/http/gateway.md#trusted-vs-trustless) for more information.
```

**Review by lidel — APPROVED** [review]
Thanks! 

I've adjusted required branch protections in this repo to require new fleek  CI, hopefullu unblocking others to merge PRs in this repo (lmk if you get blocked again).

**Pull request merged**